internal/compiler/generator/rust.rs

```diff
@@ -326,6 +330,15 @@ fn generate_shared_globals(llr: &llr::CompilationUnit) -> TokenStream {
         .map(global_inner_name)
         .collect::<Vec<_>>();
 
+    let apply_constant_scale_factor = if !compiler_config.const_scale_factor.approx_eq(&1.0) {
+        let factor = compiler_config.const_scale_factor;
+        Some(
+            quote!(adapter.window().dispatch_event(slint::platform::WindowEvent::ScaleFactorChanged{ scale_factor: #factor as f32 });),
+        )
```

Comment:
Not important, but i'd suggest to move the `as f32` in slint compilation time from rust compilation time.

```suggestion
        let factor = compiler_config.const_scale_factor as f32;
        Some(
            quote!(adapter.window().dispatch_event(slint::platform::WindowEvent::ScaleFactorChanged{ scale_factor: #factor });),
        )
```

also this `else{None}` pattern is maybe something witch could be done with `(!compiler_config.const_scale_factor.approx_eq(&1.0)).then(|| ...)`  you decide what takes the less lines :-)

Reply:
I was contemplating that pattern, but I don't find `(!cond)..then(|| ...)` quite readable :-(. Good point about the f32.